src/Microsoft.AspNetCore.Razor.Tools/MutexName.cs

```diff
@@ -12,7 +12,11 @@ public static string GetClientMutexName(string pipeName)
 
         public static string GetServerMutexName(string pipeName)
         {
-            return $"{pipeName}.server";
+            // We want to prefix this with Global\ because we want this mutex to be visible
+            // across terminal sessions which is useful for cases like shutdown.
+            // https://msdn.microsoft.com/en-us/library/system.threading.mutex(v=vs.110).aspx#Remarks
+            // This still wouldn't allow other users to access the server because the pipe will fail to connect.
+            return $"Global\\{pipeName}.server";
```

Comment:
@peterhuene I had to do this to make our shutdown command work across different terminal sessions on Unix systems. Without this the command just thinks the server is not running. 
It's possible Roslyn might have this problem as well as they do something [similar](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Server/VBCSCompiler/BuildServerController.cs#L90). Could you do a quick check if your Roslyn shutdown code works on Unix machines across terminal sessions? If not, they might have to add this prefix as well.

Reply:
Looks like it won't work for Roslyn, so we should probably open an issue there as well.  We're only making this new mechanism work for Razor currently, so hopefully we can get that addressed at the same time Roslyn starts writing pid files for us.

Reply:
Do you need us to file an issue for it in Roslyn?

Reply:
I'll do it, thanks for the heads up!